qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -256,32 +277,45 @@ def run(self, dag):
             # After all free gates are exhausted, heuristically find
             # the best swap and insert it. When two or more swaps tie
             # for best score, pick one randomly.
+
             if extended_set is None:
                 extended_set = self._obtain_extended_set(dag, front_layer)
-            swap_scores = {}
-            for swap_qubits in self._obtain_swaps(front_layer, current_layout):
-                trial_layout = current_layout.copy()
-                trial_layout.swap(*swap_qubits)
-                score = self._score_heuristic(
-                    self.heuristic, front_layer, extended_set, trial_layout, swap_qubits
+            extended_set_list = EdgeList(len(extended_set))
+            for x in extended_set:
+                extended_set_list.append(
+                    self._bit_indices[x.qargs[0]], self._bit_indices[x.qargs[1]]
+                )
+
+            front_layer_list = EdgeList(len(front_layer))
+            for x in front_layer:
+                front_layer_list.append(
+                    self._bit_indices[x.qargs[0]], self._bit_indices[x.qargs[1]]
                 )
-                swap_scores[swap_qubits] = score
-            min_score = min(swap_scores.values())
-            best_swaps = [k for k, v in swap_scores.items() if v == min_score]
-            best_swaps.sort(key=lambda x: (self._bit_indices[x[0]], self._bit_indices[x[1]]))
+            swap_candidates = self._obtain_swaps(front_layer, layout)
+            swap_scores = SwapScores(list(swap_candidates))
```

Comment:
We are, I did this just to match types because I wrote swap scores by taking a `Vec<[usize; 2]>` input on it's `__new__` method. You're right we should change that to be a set input so we can avoid this overhead. But that being said I have a couple local commits on this branch that I'm running benchmarks on now that eliminates `_obtain_swaps` from python and removes the `SwapScores` class completely and just tracks the minimum score over each iteration.